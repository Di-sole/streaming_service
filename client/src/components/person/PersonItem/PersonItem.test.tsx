import { render, screen } from '@testing-library/react';
import { actorSuggestions } from '../../../testAsserts/testItems';
import PersonItem from './PersonItem';

jest.mock('next-i18next', () => ({
    useTranslation: () => {
        return {
            t: (str: string) => str,
            i18n: {
                changeLanguage: () => new Promise(() => {}),
            },
        };
    },
}));

jest.mock('next/router', () => ({
    useRouter() {
        return {
            pathname: '',
        };
    },
}));

describe('PERSONITEM SNAPSHOTS TESTS', () => {
    test('large PersonItem should not be changed', () => {
        render(<PersonItem person={actorSuggestions[0]} size="large" />);
        const link = screen.getByRole('link');
        expect(link).toMatchSnapshot();
    });

    test('medium PersonItem should not be changed', () => {
        render(<PersonItem person={actorSuggestions[0]} size="medium" />);
        const link = screen.getByRole('link');
        expect(link).toMatchSnapshot();
    });

    test('small PersonItem should not be changed', () => {
        render(<PersonItem person={actorSuggestions[0]} size="small" />);
        const link = screen.getByRole('link');
        expect(link).toMatchSnapshot();
    });
});
