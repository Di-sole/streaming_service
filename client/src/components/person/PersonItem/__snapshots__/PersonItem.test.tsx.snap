// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PERSONITEM SNAPSHOTS TESTS large PersonItem should not be changed 1`] = `
<a
  class="container container_large "
  href="/persons/71427"
>
  <div
    class="container__imageContainer"
  >
    <div
      class="container__imageWrapper"
    >
      <img
        alt="Франсуа Клюзе François Cluzet"
        class="container__image"
        data-nimg="1"
        decoding="async"
        height="300"
        loading="lazy"
        src="/_next/image?url=https%3A%2F%2Fst.kp.yandex.net%2Fimages%2Factor_iphone%2Fiphone360_71427.jpg&w=640&q=75"
        srcset="/_next/image?url=https%3A%2F%2Fst.kp.yandex.net%2Fimages%2Factor_iphone%2Fiphone360_71427.jpg&w=384&q=75 1x, /_next/image?url=https%3A%2F%2Fst.kp.yandex.net%2Fimages%2Factor_iphone%2Fiphone360_71427.jpg&w=640&q=75 2x"
        style="color: transparent;"
        width="300"
      />
    </div>
    <div
      class="container__amountBadge"
    >
      1
    </div>
  </div>
  <div
    class="container__textContainer"
  >
    <h2
      class="container__fullName"
    >
      François
      <br />
      Cluzet
    </h2>
    <p
      class="container__amountMovies"
    >
      1
       
      filmography.singleMovie
    </p>
  </div>
</a>
`;

exports[`PERSONITEM SNAPSHOTS TESTS medium PersonItem should not be changed 1`] = `
<a
  class="container container_medium "
  href="/persons/71427"
>
  <div
    class="container__imageContainer"
  >
    <div
      class="container__imageWrapper"
    >
      <img
        alt="Франсуа Клюзе François Cluzet"
        class="container__image"
        data-nimg="1"
        decoding="async"
        height="300"
        loading="lazy"
        src="/_next/image?url=https%3A%2F%2Fst.kp.yandex.net%2Fimages%2Factor_iphone%2Fiphone360_71427.jpg&w=640&q=75"
        srcset="/_next/image?url=https%3A%2F%2Fst.kp.yandex.net%2Fimages%2Factor_iphone%2Fiphone360_71427.jpg&w=384&q=75 1x, /_next/image?url=https%3A%2F%2Fst.kp.yandex.net%2Fimages%2Factor_iphone%2Fiphone360_71427.jpg&w=640&q=75 2x"
        style="color: transparent;"
        width="300"
      />
    </div>
  </div>
  <div
    class="container__textContainer"
  >
    <h2
      class="container__fullName"
    >
      François
      <br />
      Cluzet
    </h2>
    <p
      class="container__amountMovies"
    >
      1
       
      filmography.singleMovie
    </p>
  </div>
</a>
`;

exports[`PERSONITEM SNAPSHOTS TESTS small PersonItem should not be changed 1`] = `
<a
  class="container container_small "
  href="/persons/71427"
>
  <div
    class="container__imageContainer"
  >
    <div
      class="container__imageWrapper"
    >
      <img
        alt="Франсуа Клюзе François Cluzet"
        class="container__image"
        data-nimg="1"
        decoding="async"
        height="300"
        loading="lazy"
        src="/_next/image?url=https%3A%2F%2Fst.kp.yandex.net%2Fimages%2Factor_iphone%2Fiphone360_71427.jpg&w=640&q=75"
        srcset="/_next/image?url=https%3A%2F%2Fst.kp.yandex.net%2Fimages%2Factor_iphone%2Fiphone360_71427.jpg&w=384&q=75 1x, /_next/image?url=https%3A%2F%2Fst.kp.yandex.net%2Fimages%2Factor_iphone%2Fiphone360_71427.jpg&w=640&q=75 2x"
        style="color: transparent;"
        width="300"
      />
    </div>
  </div>
  <div
    class="container__textContainer"
  >
    <h2
      class="container__fullName"
    >
      François
      <br />
      Cluzet
    </h2>
    <h3
      class="container__role"
    >
      Actor
    </h3>
  </div>
</a>
`;
